File: 01 - Classical/02 - Fibonachi/fibonacci_last_digit.py
# Uses python3
import numpy as np

def get_fibonacci_last_digit_naive(n):
    
    if n <= 1:
        return n
    numbers=np.zeros(n+1)
    for i in range(n+1):
        if i<=1:
            numbers[i]=i
        else:
            numbers[i]=int((numbers[i-1]+numbers[i-2])%10)
        
    return int(numbers[n])

def test():
    
    assert(get_fibonacci_last_digit_naive(3)==2)
    assert(get_fibonacci_last_digit_naive(327305)==5)
    assert(get_fibonacci_last_digit_naive(331)==9)
    
    print('All tests are passed')

if __name__ == '__main__':
    
    n = int(input())
    test()
    print(get_fibonacci_last_digit_naive(n))

File: 01 - Classical/02 - Fibonachi/fibonacci_last_digit.cpp
#include <iostream>
#include <vector>

int get_fibonacci_last_digit_naive(int n) {
    if (n <= 1){
       return n;
    }
        

    std::vector<int> numbers(n);

    for (int i=0; i<(n+1); i++){
        if (i==0){
            numbers[i]=0;
        }
        else if (i==1){
            numbers[i]=1;
        }
        else{
            numbers[i]=(numbers[i-2]+numbers[i-1])%10;
        }
        

    }

   

    return numbers[n];
}

int main() {
    int n;
    std::cin >> n;
    int c = get_fibonacci_last_digit_naive(n);
    std::cout << c << '\n';
    }
